checks/raw/fuzzing.go

```diff
@@ -91,3 +135,101 @@ func checkOSSFuzz(c *checker.CheckRequest) (bool, error) {
+	// To get the prominent programming language(s) to be checked.
+	langMap, err := c.RepoClient.ListProgrammingLanguages()
+	if err != nil {
+		return false, nil, fmt.Errorf("get programming languages of repo failed %w", err)
+	}
+	langsProminent, err := getProminentLanguages(langMap)
+	if err != nil {
+		return false, nil, fmt.Errorf("error when getting promiment languages: %w", err)
+	}
+
+	data := filesWithPatternStr{
+		files: make([]checker.File, 0),
+	}
+
+	// Iterate the prominant language list and check for fuzz funcs per language.
+	for _, lang := range *langsProminent {
+		// Search language fuzz patterns in the hashmap.
+		pattern, found := languageFuzzSpecsMap[lang]
+		if !found {
+			// Fuzz patterns for the current language not supported yet.
+			continue
+		}
+		// Get patterns for file and func.
+		filePattern, funcPattern := pattern.fuzzFileMatchPattern, pattern.fuzzFuncRegexPattern
+		matcher := fileparser.PathMatcher{
+			Pattern:       filePattern,
+			CaseSensitive: false,
+		}
+		data.pattern = funcPattern
+		err = fileparser.OnMatchingFileContentDo(c.RepoClient, matcher, getFuzzFunc, &data)
+		if err != nil {
+			return false, nil, fmt.Errorf("error when OnMatchingFileContentDo: %w", err)
+		}
+	}
+	if len(data.files) == 0 {
+		return false, nil, nil
+	}
+	return true, data.files, nil
+}
+
+// This is the callback func for interface OnMatchingFileContentDo
+// used for matching fuzz functions in the file content,
+// and return a list of files (or nil for not found).
+var getFuzzFunc fileparser.DoWhileTrueOnFileContent = func(path string, content []byte, args ...interface{}) (bool, error) {
+	if len(args) != 1 {
+		return false, fmt.Errorf("getFuzzFunc requires exactly one argument: %w", errInvalidArgLength)
+	}
+	pdata, ok := args[0].(*filesWithPatternStr)
+	if !ok {
+		return false, fmt.Errorf("invalid arg type: %w", errInvalidArgType)
+	}
+	r, _ := regexp.Compile(pdata.pattern)
+	lines := bytes.Split(content, []byte("\n"))
+	for i, line := range lines {
+		found := r.FindString(string(line))
+		if found != "" {
+			// If fuzz func is found in the file, add it to the file array,
+			// with its file path as Path, func name as Snippet,
+			// FileTypeFuzz as Type, and # of lines as Offset.
+			pdata.files = append(pdata.files, checker.File{
+				Path:    path,
+				Type:    checker.FileTypeSource,
+				Snippet: found,
+				Offset:  uint(i + 1), // Since the # of lines starts from zero.
+			})
+		}
+	}
+	return true, nil
+}
+
+func getProminentLanguages(langs map[string]int) (*[]string, error) {
+	if langs == nil {
+		return nil, fmt.Errorf("no languages found in map")
+	}
+	numLangs := len(langs)
+	totalLoC := 0
+	for _, LoC := range langs {
+		totalLoC += LoC
+		numLangs++
+	}
+	// Var avgLoC calculates the average lines of code in the current repo,
+	// and it can stay as an int, no need for a float value.
+	avgLoC := totalLoC / numLangs
+
+	// Languages that has lines of code above average will be considered prominent.
```

Comment:
nit: has -> have

Reply:
np, maybe I should use an English grammar check tool :P